code/game/gamemodes/antag_spawner.dm

```diff
@@ -53,15 +53,18 @@
 	if(loc == H || (in_range(src, H) && istype(loc, /turf)))
 		H.set_machine(src)
 		if(href_list["school"])
-			if (used)
+			if(used)
 				H << "You already used this contract!"
 				return
-			var/list/candidates = pollCandidates("Do you want to play as a wizard's [href_list["school"]] apprentice?", ROLE_WIZARD, null, ROLE_WIZARD, 150)
+			var/list/candidates = pollCandidatesForMob("Do you want to play as a wizard's [href_list["school"]] apprentice?", ROLE_WIZARD, null, ROLE_WIZARD, 150, src)

```

Comment:
That's why I'm using pollCandidatesForMob with the spawn item itself as an argument; it makes sure the mob("mob" it happens to be an item but that doesn't matter) actually exists after the sleep.
I'll adjust that, instead, to check if the 'mob' has a loc.